### PR TITLE
Try fix MultipleTypeField: Add post_inits metric

### PR DESCRIPTION
**Pull request by gpotter2**
fixes https://github.com/secdev/scapy/issues/1484

Is certainly improvable, any ideas will be appreciated

### CODE REVIEW

File: scapy/layers/dns.py [hunk 1]
```diff
@@ -43,6 +43,8 @@ def h2i(self, pkt, x):
     def i2m(self, pkt, x):
         if x == b".":
             return b"\x00"
+        elif isinstance(x, str):
```
Comment (p-l-) [inline scapy/layers/dns.py]: I'm not sure when/if this can happen, as "internal" value should always be bytes.

File: scapy/asn1fields.py [hunk 2]
```diff
@@ -37,6 +37,7 @@ class ASN1F_element(object):
 class ASN1F_field(ASN1F_element):
     holds_packets = 0
     islist = 0
+    post_init = False
```
Comment (p-l-) [inline scapy/asn1fields.py]: Why?

**Comment by gpotter2** [comment]
I don't really like myself the extra layer of complexity that this PR adds :/ I really would have like a different fix. If you have time to look into it, it would be great

File: scapy/packet.py [hunk 3]
```diff
@@ -127,6 +131,7 @@ def __init__(self, _pkt=b"", post_transform=None, _internal=0, _underlayer=None,
         self.fieldtype = {}
         self.packetfields = []
         self.payload = NoPayload()
+        self.post_inits = []
         self.init_fields()
```
Comment (gpotter2) [inline scapy/packet.py]: NOTE TO SELF: just order the fields following the fields_desc order during initialization. That will fix the problem way easier.

Use stg like
```
for f in fields_desc:
    v = kwargs.get(v, None)
    if v:
        self.setfieldval.....
```

Comment (gpotter2) [reply scapy/packet.py]: FTR, PR moved to https://github.com/secdev/scapy/pull/1550

**Pull request closed without merging**